Sources/protoc-gen-swift/StringUtils.swift

```diff
@@ -274,3 +274,12 @@ func stringToEscapedStringLiteral(_ s: String) -> String {
   }
   return out + "\""
 }
+
+func isValidSwiftIdentifier(_ s: String) -> Bool {
```

Comment:
I would like to use `isValidSwiftIdentifier` throughout the generator.  How hard would it be to expand it to actually do what it says?

The full spec is not small, but it's not inherently complicated, either.  It boils down to "first character must be this and the rest must be that":

 https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/zzSummaryOfTheGrammar.html